internal/tui/components/chat/messages/renderer.go

```diff
@@ -886,7 +899,8 @@ func (tr agentRenderer) Render(v *toolCallCmp) string {
 	prompt = strings.ReplaceAll(prompt, "\n", " ")
 
 	header := tr.makeHeader(v, "Agent", v.textWidth())
-	if res, done := earlyState(header, v); v.cancelled && done {
+	// TODO: revisit logic, now that we have more ToolCallStates
+	if res, done := renderStatusOnly(header, v); v.call.State == enum.ToolCallStateCancelled && done {
 		return res
 	}
```

Comment:
This neededs at least a better name.
I still do not like that `renderStatusOnly` even returns a `bool` in the first place.

Reply:
Still worth reviewing again.